### PR TITLE
Package dev-libs/xmlrpc-c 1.59.03

### PR DESCRIPTION
**Pull request by andrewshadura**
Update dev-libs/xmlrpc-c to 1.59.03

Depends on PR #39396.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: dev-libs/xmlrpc-c/xmlrpc-c-1.59.03.ebuild [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		-e "/CFLAGS_COMMON/s|-g -O3$||" \
+		-e "/CXXFLAGS_COMMON/s|-g$||" \
+		common.mk || die
+	eautoconf
+}
+
+src_configure() {
+	tc-export PKG_CONFIG
+
+	append-ldflags -Wl,-z,now
```
Comment (thesamesam) [inline dev-libs/xmlrpc-c/xmlrpc-c-1.59.03.ebuild:56]: I think this merits more rationale. It's already the default in 23.0 profiles.

Comment (andrewshadura) [reply dev-libs/xmlrpc-c/xmlrpc-c-1.59.03.ebuild:56]: Ok, excluding this change.

Comment (andrewshadura) [reply dev-libs/xmlrpc-c/xmlrpc-c-1.59.03.ebuild]: I have re-pushed the change without this bit.
However, when building with the default profiles, I didn’t see `-Wl,-z,now` in the logs, so apparently it doesn’t get picked up for some reason.

Comment (thesamesam) [reply dev-libs/xmlrpc-c/xmlrpc-c-1.59.03.ebuild]: The GCC is instructed to always pass it via a patch (or in the case of Clang, a configuration file). so you don't see it.

**Pull request closed without merging**